src/goto-programs/character_refine_preprocess.cpp

```diff
@@ -0,0 +1,2564 @@
+  return or_exprt(
+    or_exprt(condition0, condition1),
+    or_exprt(condition2, or_exprt(condition3, condition4)));
+}
+
+/*******************************************************************\
+
+Function: character_refine_preprocesst::convert_is_whitespace_char
+
+  Inputs:
+    target - a position in a goto program
+
+ Purpose: Converts function call to an assignment of an expression
+          corresponding to the java method Character.isWhitespace:(C)Z
+
+\*******************************************************************/
+
+void character_refine_preprocesst::convert_is_whitespace_char(
+  conversion_input &target)
+{
+  convert_char_function(
+    &character_refine_preprocesst::expr_of_is_whitespace, target);
+}
+
+/*******************************************************************\
+
+Function: character_refine_preprocesst::convert_is_whitespace_int
+
+  Inputs:
+    target - a position in a goto program
+
+ Purpose: Converts function call to an assignment of an expression
+          corresponding to the java method Character.isWhitespace:(I)Z
+
+\*******************************************************************/
+
+void character_refine_preprocesst::convert_is_whitespace_int(
+  conversion_input &target)
+{
+  convert_is_whitespace_char(target);
+}
+
+/*******************************************************************\
+
+Function: character_refine_preprocesst::expr_of_low_surrogate
+
+  Inputs:
+    expr - An expression of type character
+    type - A type for the output
+
+ Outputs: A integer expression of the given type
+
+ Purpose: Returns the trailing surrogate (a low surrogate code unit)
+          of the surrogate pair representing the specified
+          supplementary character (Unicode code point) in the UTF-16
+          encoding.
+
+\*******************************************************************/
+
+exprt character_refine_preprocesst::expr_of_low_surrogate(
+  const exprt &chr, const typet &type)
+{
+  exprt uDC00=from_integer(0xDC00, type);
+  exprt u0400=from_integer(0x0400, type);
+  return plus_exprt(uDC00, mod_exprt(chr, u0400));
+}
+
+/*******************************************************************\
+
+Function: character_refine_preprocesst::convert_low_surrogate
+
+  Inputs:
+    target - a position in a goto program
+
+ Purpose: Converts function call to an assignment of an expression
+          corresponding to the java method Character.lowSurrogate:(I)Z
+
+\*******************************************************************/
+
+void character_refine_preprocesst::convert_low_surrogate(
+  conversion_input &target)
+{
+  convert_char_function(
+    &character_refine_preprocesst::expr_of_low_surrogate, target);
+}
+
+/*******************************************************************\
+
+Function: character_refine_preprocesst::expr_of_reverse_bytes
+
+  Inputs:
+    expr - An expression of type character
+    type - A type for the output
+
+ Outputs: A character expression of the 
```

Comment:
Is there any way to catch or check additional function names (think about evolution of the Java SDK)? I'm thinking along the lines of "if the function call has prefix java::java.lang.Character." then it must be in the `conversion_table` or in some (to be created) `conversion_not_supported` table. Would that seem reasonable to do?